packages/import/src/index.ts

```diff
@@ -360,9 +198,246 @@ function visitFile(
       }
     }
   }
-  return visitedFiles.get(filePath);
+  return visitedFiles.get(filePath)!;
+}
+
+export function extractDependencies(
+  filePath: string,
+  fileContents: string
+): {
+  definitionsByName: Map<string, Set<DefinitionNode>>;
+  dependenciesByDefinitionName: Map<string, Set<string>>;
+} {
+  const definitionsByName = new Map<string, Set<DefinitionNode>>();
+  const dependenciesByDefinitionName = new Map<string, Set<string>>();
+
+  const { document } = parseGraphQLSDL(filePath, fileContents, {
+    noLocation: true,
+  });
+
+  for (const definition of document.definitions) {
+    visitDefinition(definition, definitionsByName, dependenciesByDefinitionName);
+  }
+
+  return {
+    definitionsByName,
+    dependenciesByDefinitionName,
+  };
+}
+
+function visitDefinition(
+  definition: DefinitionNode,
+  definitionsByName: Map<string, Set<DefinitionNode>>,
+  dependenciesByDefinitionName: Map<string, Set<string>>
+): void {
+  // TODO: handle queries without names
```

Comment:
Looking at the logic here I'm pretty sure that queries without names aren't handled here.